forui/lib/src/widgets/tooltip.dart

```diff
@@ -0,0 +1,396 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter/widgets.dart';
+
+import 'package:meta/meta.dart';
+
+import 'package:forui/forui.dart';
+
+/// A controller that controls whether a [FPopover] is shown or hidden.
+final class FTooltipController extends ChangeNotifier {
+  static final _fadeTween = Tween<double>(begin: 0, end: 1);
+  static final _scaleTween = Tween<double>(begin: 0.95, end: 1);
+
+  final OverlayPortalController _overlay = OverlayPortalController();
+  late final AnimationController _animation;
+  late final Animation<double> _fade;
+  late final Animation<double> _scale;
+
+  /// Creates a [FTooltipController] with the given [vsync] and animation [duration].
+  FTooltipController({required TickerProvider vsync, duration = const Duration(milliseconds: 100)}) {
+    _animation = AnimationController(vsync: vsync, duration: duration);
+    _fade = _fadeTween.animate(_animation);
+    _scale = _scaleTween.animate(_animation);
+  }
+
+  /// Convenience method for toggling the current [shown] status.
+  ///
+  /// This method should typically not be called while the widget tree is being
+  /// rebuilt.
+  Future<void> toggle() async => shown ? hide() : show();
+
+  /// Shows the tooltip.
+  ///
+  /// If [shown] is already true, calling this method brings the tooltip it controls to the top.
+  ///
+  /// This method should typically not be called while the widget tree is being rebuilt.
+  Future<void> show() async {
+    _overlay.show();
+    await _animation.forward();
+    notifyListeners();
+  }
+
+  /// Hides the tooltip.
+  ///
+  /// Once hidden, the tooltip will be removed from the widget tree the next time the widget tree rebuilds, and stateful
+  /// widgets in the tooltip may lose states as a result.
+  ///
+  /// This method should typically not be called while the widget tree is being rebuilt.
+  Future<void> hide() async {
+    await _animation.reverse();
+    _overlay.hide();
+    notifyListeners();
+  }
+
+  /// True if the tooltip is currently being shown. False if it is hidden.
+  bool get shown => _overlay.isShowing;
+
+  @override
+  void dispose() {
+    _animation.dispose();
+    super.dispose();
+  }
+}
+
+/// The tooltip's behavior.
+enum FToolTipBehavior {
+  /// Denotes that the tooltip is shown when hovered over.
+  hover,
+
+  /// Denotes that the tooltip is shown when hovered over or long pressed.
+  hoverOrLongPress,
+
+  /// Denotes that the tooltip is shown when long pressed.
+  longPress,
+}
```

Comment:
Is using 2 boolean cleaner?

Reply:
I personally think that using 2 booleans is not a good idea since it's possible to pass in `false` for both while also not passing it a controller to manually show the tooltip.

Reply:
Actually isn't that a great idea, we don't need the manual constructor that way.